packages/sprintf/index.js

```diff
@@ -53,26 +53,19 @@ var PATTERN =
  * sprintf( 'Hello %s!', 'world' );
  * // ⇒ 'Hello world!'
  * ```
- *
- * @param {string}                                   string printf format string
- * @param {...string|string[]|Object<string,string>} [args] String arguments.
+ * @template {string} T
+ * @param {T} string - string printf format string
+ * @param {import('./types/index.d').SprintfArgs<T>|import('./types/index.d').SprintfArgs<T>[]} args String arguments.
  *
  * @return {string} Formatted string.
  */
-export default function sprintf(string, args) {
-	var i;
+export default function sprintf(string, ...args) {
+	var i = 0;
 
-	if (!Array.isArray(args)) {
-		// Construct a copy of arguments from index one, used for replace
```

Comment:
This is nice to be able to able to drop the array cloning!

Reply:
I would like to point out that I have also simplified the iterator variable (`var i = 0;`). This is because there was no longer any need to 'reset' after creating the cloned array, so we can start from `0` and not "correct" subsequent values. (if it seems risky, let me know, I'll restore it as before)

Reply:
Nope, that sounds like a sensible change! Let's keep it.